core/codenvy-hosted-api-permission/src/main/java/com/codenvy/api/permission/server/PermissionManager.java

```diff
@@ -0,0 +1,158 @@
+/*
+ *  [2012] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.api.permission.server;
+
+import com.codenvy.api.permission.server.dao.PermissionsStorage;
+
+import org.eclipse.che.api.core.ConflictException;
+import org.eclipse.che.api.core.ServerException;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Class that holds supported {@link PermissionsStorage} and delegates calls to them
```

Comment:
I think you can improve this doc. Information that it holds PermissionsStorage doesn't make sense to clients of that class. you can describe it as class that operates with permissions